### PR TITLE
New endpoint for ALL repos a user can act on

### LINKED ISSUE #42: Endpoint for listing all user github repos

### PR DESCRIPTION
**Pull request by oudeismetis**
fixes #42

### CODE REVIEW

File: franklin/github/views.py [hunk 1]
```diff
@@ -130,14 +132,50 @@ def register_repo(request):
...
+
+    responseMessages:
+        - code: 500
+          message: Error from Github.
+    """
+    if request.method == 'GET':
+        # TODO - in the model, github response should map to a serializer which
+        # we should use here to define the respone type
+        github_repos = request.user.details.get_user_repos()
+        # TODO - return error from github
```
Comment (oudeismetis) [inline franklin/github/views.py]: Was tempted to fix these TODO items now, but that would take me down the path of refactoring a lot of things. Would prefer to wait until the last week before the holiday to kill a lot of these lingering code quality items.

File: franklin/core/urls.py [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
+from github.views import deploy_hook, deployed_repos, deployable_repos, \
+    register_repo, get_auth_token
 
 urlpatterns = [
     url(r'^auth/github/$', get_auth_token, name='get_token'),
     url(r'^deployed/$', deploy_hook, name='deploy'),
     url(r'^register/$', register_repo, name='register'),
-    url(r'^userrepos/$', get_user_repos, name='userrepos'),
+    url(r'^user/repos/deployed/$', deployed_repos, name='deployed_repos'),
+    url(r'^user/repos/deployable/$', deployable_repos, name='deployable_repos'),
```
Comment (oudeismetis) [inline franklin/core/urls.py]: @camiriveron Notice the new endpoint and the remapping of the existing one.

**Comment by JulianGindi** [comment]
:+1:

**Comment by camiriveron** [comment]
image

**Pull request merged**